WMDE_FR_2025_Mobile_EN_01

Sperling-0 opened the pull request:
- Both banners

    - are based on the variant banner of the last 2024 setup.
    - has the same slide interval from DE Mobile
    - do not offer SMS as a payment channel on the Slides and in the collapsed banner
    - do not use the soft close feature.
    - have the "I already donated" link displayed somewhere on the Slides (maybe at the left bottom after moving the payment icons a little bit to the right)
    - do not have the "Where does my money go"" link on the collapsed banner
    - uses the imperative "Please make it €x yearly!" in the upsell

- Variant banner
    - uses our current copy

Ticket: https://phabricator.wikimedia.org/T405916

Sperling-0 committed:
Prepare campaign for C25_WMDE_Mobile_EN_01

Ticket: https://phabricator.wikimedia.org/T405916

Sperling-0 committed:
Implement CTRL for WMDE_FR_2025_Mobile_EN_01

- Both banners:
   - are based on the variant banner of the last 2024 setup.
   - has the same slide interval from DE Mobile
   - do not offer SMS as a payment channel on the Slides and in the collapsed banner
   - do not use the soft close feature.
   - have the "I already donated" link displayed somewhere on the Slides (maybe at the left bottom after moving the payment icons a little bit to the right)
   - do not have the "Where does my money go"" link on the collapsed banner
   - uses the imperative "Please make it €x yearly!" in the upsell

- Update tests, a new class is used in the mini banner header UoF
- Update content

Ticket: https://phabricator.wikimedia.org/T405916

Sperling-0 committed:
Prepare VAR for WMDE_FR_2025_Mobile_EN_01

Ticket: https://phabricator.wikimedia.org/T405916

Sperling-0 committed:
Implement VAR for WMDE_FR_2025_Mobile_EN_01

Ticket: https://phabricator.wikimedia.org/T405916

Sperling-0 committed:
Fix progress bar to display corrent amount

- Fix broken css
- Increase banner height as CTRL banner has too much text in the last slide

Ticket: https://phabricator.wikimedia.org/T405916

gbirke commented on banners/mobile_english/WMDE_FR_2025_Mobile_EN_01/components/MiniBanner.vue:
```diff
@@ -0,0 +1,58 @@
+				</div>
+			</div>
+
+		</div>
```
**suggestion:** more semantic class names & removal of unneccessary wrappers

These class names signify what's inside them

The `div` wrapper should not be neessary

```suggestion
			<div class="wmde-banner-mini-footer">

					<button
						class="wmde-banner-mini-already-donated-button"
						@click.prevent="$emit( 'alreadyDonatedClicked' )"
					>
						<span class="wmde-banner-mini-already-donated-icon"><SimpleInfoIcon/></span>
						<span>{{ $translate( 'mini-banner-already-donated-button' ) }}</span>
					</button>

				<div class="wmde-banner-mini-payment-icons">
					<PayPalLogo/>
					<VisaLogo/>
					<MastercardLogo/>
				</div>
			</div>

		</div>
```

Sperling-0 replied:
@gbirke Yes, I should have used more semantic class names. I have corrected them now.

gbirke commented on banners/mobile_english/WMDE_FR_2025_Mobile_EN_01/styles/MiniBanner.scss:
```diff
@@ -0,0 +1,142 @@
+				}
+			}
+		}
+	}
```
**suggestion:** less nesting & preserving the previous definitions

Using names like I suggested for the markup will help defining each element as its own.

The `justify-content` property was used to create some space between the payment icons, so and does not make much sense for the footer itself, which just creates 2 columns.


```suggestion
		&-footer {
			display: flex;
			height: 45px;
			padding: 10px 15px;
			align-items: center;
	}
	&-payment-icons {
			display: flex;
			justify-content: space-between;
			align-items: center;
			
			svg {
				height: 18px;
				width: auto;
				margin-left: 10px;
			}
	}
```

Sperling-0 replied:
@gbirke Following are the screenshots which show two versions of banners, one version has my css another has what you proposed in the comment above:

|| Sperling-0    | Gabriel |
| -------- | -------- | ------- |
|Portrait mode| <img width="1788" height="1032" alt="Screenshot from 2025-10-08 09-51-08" src="https://github.com/user-attachments/assets/8bd528ed-5b6d-42f6-810b-d260b9c979cf" /> | <img width="1788" height="1032" alt="Screenshot from 2025-10-08 09-50-38" src="https://github.com/user-attachments/assets/1bb72fe8-1007-47a9-a518-d414cb5080a5" /> |
| Landscape mode | <img width="1788" height="1032" alt="Screenshot from 2025-10-08 09-49-29" src="https://github.com/user-attachments/assets/3e51162d-cec9-4d72-87c8-cff9247f4b2d" /> | <img width="1788" height="1032" alt="Screenshot from 2025-10-08 09-49-58" src="https://github.com/user-attachments/assets/6a28035f-6fd3-4cc2-8a7e-b306eb73764b" /> |

Sperling-0 committed:
Use semantic class names for css

Ticket: https://phabricator.wikimedia.org/T405916

gbirke commented on banners/mobile_english/WMDE_FR_2025_Mobile_EN_01/components/MiniBanner.vue:
```diff
@@ -0,0 +1,58 @@
+						<span>{{ $translate( 'mini-banner-already-donated-button' ) }}</span>
+					</button>
+				</div>
+				<div class="payment-icons">
```
**nitpick:** missing prefix

We usually prefix almost all classes with `wmde-banner` (and in this case `wmde-banner-mini` to avoid name clashes with mediawiki styles. You can then also change the CSS to avoid nesting.

Sperling-0 replied:
@gbirke I fixed it, let me know if something is still missing.

gbirke reviewed (approved):
A small nitpick, but I'd be ok with leaving the code like this

Abban commented on banners/mobile_english/WMDE_FR_2025_Mobile_EN_01/components/BannerCtrl.vue:
```diff
@@ -0,0 +1,206 @@
+			@show-full-page-banner="onshowFullPageBanner"
+			@show-full-page-banner-preselected-amount="onShowFullPageBannerPreselectedAmount"
+			@show-use-of-funds="onShowFundsModal( 'MiniBanner' )"
+			@already-donated-clicked="onClose( 'AlreadyDonated', CloseChoices.AlreadyDonated )"
```
```suggestion
			@already-donated-clicked="onClose( 'MiniBanner', CloseChoices.AlreadyDonated )"
```

Abban commented on banners/mobile_english/WMDE_FR_2025_Mobile_EN_01/components/BannerVar.vue:
```diff
@@ -0,0 +1,206 @@
+			@show-full-page-banner="onshowFullPageBanner"
+			@show-full-page-banner-preselected-amount="onShowFullPageBannerPreselectedAmount"
+			@show-use-of-funds="onShowFundsModal( 'MiniBanner' )"
+			@already-donated-clicked="onClose( 'AlreadyDonated', CloseChoices.AlreadyDonated )"
```
```suggestion
			@already-donated-clicked="onClose( 'MiniBanner', CloseChoices.AlreadyDonated )"
```

Abban reviewed (changes requested):
I just noticed that the mobile banners are sending the wrong feature when a donor clicks "I already donated".

These changes will also make the tests fail and you'll need to adjust the `expectsEmitsCloseEventOnAlreadyDonated` test in https://github.com/wmde/fundraising-banners/blob/a20eb16ef878a0c8d1652c42e0d96c99c18d44ef/test/features/MiniBanner.ts

Sperling-0 committed:
"Already donated" shows the checkmark icon instead of the info icon

Ticket: https://phabricator.wikimedia.org/T405916

Sperling-0 committed:
Use correct 'feature' for already donated event

Ticket: https://phabricator.wikimedia.org/T405916

Sperling-0 commented:
> [!NOTE]  
> This banner uses correct 'feature' for already donated event.

Sperling-0 committed:
Use correct 'feature' for already donated event for previous banners

- Update content

Ticket: https://phabricator.wikimedia.org/T405916

Sperling-0 committed:
Donation target visible for Mini-banner

- For VAR banner the progress bar on the 1st slide is showing the donation target at the top right (as the one in the expanded banner does)
- The CTRL banner does not show it as the donation target is shown in the headline itself
- Both banners have same size
- The VAR banner has gap between headline and progress-bar with donation amount visible at top right

Ticket: https://phabricator.wikimedia.org/T405916

moiikana commented on banners/mobile_english/WMDE_FR_2025_Mobile_EN_01/content/BannerSlidesVar.vue:
```diff
@@ -1,31 +1,30 @@
-			If you can, please join the 1% of readers who give. <span class="wmde-banner-text-animated-highlight">If
-				everyone reading this right now gave just €10, we'd hit our goal in a couple of hours.</span>
-			It's hard to know what to trust online these days. Disinformation and scammers are everywhere.
+			Most people donate because they find Wikipedia useful. The average donation is €22.89, but even €5 helps us.
```
there is a €22.89 here, this should be using a template variable instead

Sperling-0 replied:
Fixed this in Slides and Text file.
